zpy/client_util.py

```diff
@@ -123,3 +139,259 @@ def clear_last_print():
 def is_done(state: str):
     """Returns True if state is a done state, false otherwise."""
     return state in ["READY", "CANCELLED", "PACKAGING_FAILED", "GENERATING_FAILED"]
+
+
+# def dict_to_list(d: dict) -> list:
+#     """Converts dict to list"""
+#     return list(dict(d).values())
+
+
+def remove_n_extensions(path: Union[str, Path], n: int) -> Path:
+    """
+    Removes n extensions from the end of a path. Example: "image.rgb.png" becomes "image" for n = 2
+    Args:
+        path: Path.
+    Returns:
+        Path: Path minus n extensions.
```

Comment:
if you want it to be a Path object, you could try something like this?
```
for _ in range(n):
    p = p.with_suffix('')
```
adapted from https://discuss.python.org/t/correct-way-to-remove-all-file-extensions-from-a-path-with-pathlib/6711

Reply:
got it switched over. Having a test for this already was handy

Reply:
also kept it as returning a string. Was debating path, but it was prefixing everything with a "." which caused it to fail tests. Didn't want to put a "." in front of the image "relative_path" in the json

Reply:
Love to hear it. This is the dream for tests - they don't care about the implementation, just the result, so you can swap out implementation and as long as the test passes, you can be confident the code that depends on it will still work.